lektor/admin/static/js/widgets/ToastEditor.jsx

```diff
@@ -0,0 +1,213 @@
+'use strict'
+
+import React from 'react'
+import Editor from 'tui-editor'
+
+import Component from '../components/Component'
+
+import 'codemirror/lib/codemirror.css'
+import 'tui-editor/dist/tui-editor.min.css'
+import 'tui-editor/dist/tui-editor-contents.min.css'
+
+import LektorLinkExtension from './tuiEditorLektorLinkExt.jsx'
+import LektorImageExtension from './tuiEditorLektorImageExt.jsx'
+
+const toolbarItems = [
+  'heading',
+  'bold',
+  'italic',
+  'strike',
+  'divider',
+  'hr',
+  'quote',
+  'divider',
+  'ul',
+  'ol',
+  // 'task',
+  'indent',
+  'outdent',
+  'divider',
+  'table',
+  // 'image',
+  // 'link',
+  'divider',
+  'code',
+  'codeblock'
+]
+
+class ToastEditor extends Component {
+  constructor () {
+    super()
+    this.element = null
+    this.count = 0
+    this.content = null
+
+    this.onChange = this.onChange.bind(this)
+    this.onLoad = this.onLoad.bind(this)
+    this.onRef = this.onRef.bind(this)
+    this.blur = this.blur.bind(this)
+    this.focus = this.focus.bind(this)
+
+    this.state = {
+      style: {},
+      editor: null
+    }
+  }
+
+  // from http://youmightnotneedjquery.com/#outer_height_with_margin
+  outerHeight (el) {
```

Comment:
Since this does not actually have anything to do with the class (doesn't need `this`) I think it should be a free function. Maybe it should be moved to `utils.jsx`?

Reply:
I can actually remove this function as this was added before I realised that the admin UI already uses jQuery, making this function redundant.